R/parameters.R

```diff
@@ -19,18 +19,41 @@ get_parameters <- function(iso3c = NULL,
     contact_matrix_set <- squire::get_mixing_matrix(iso3c = iso3c)
   }
 
-  c(
-    squire::parameters_explicit_SEEIR(
+  parameters <- list(
+    sq = squire::parameters_explicit_SEEIR(
       population = population,
       country = get_country(iso3c),
       contact_matrix_set = contact_matrix_set,
-      dt = 1, # dt should always be 1 as individual is always discrete time
+      dt = 1, # dt should always be 1 as individual is always a discrete time
       time_period = time_period,
       ...
     ),
     time_period = time_period,
     max_age = max_age
   )
+
+  parstemp <- parameters$sq
+
+  if (is.null(parameters$max_age)) {
+    newlist <- list(time_period = parameters$time_period, max_age = 100)
+    parstemp <- append(parstemp, newlist)
+
+  }
+  else {
+    newlist <- list(time_period = parameters$time_period,
+                     max_age = parameters$max_age)
+    parstemp <- append(parstemp, newlist)
+  }
+
+  #Add asymptomatic parameters
+  pars_asymp <- get_asymptomatic()
+
+  pars <- append(parstemp, pars_asymp)
+
+  # Remove contact _matrix_set from parameter list as this is passed straight
+  # into infection_process
+  pars <- remove_non_numerics(pars)
```

Comment:
It's not clear that the return value is intentional.

Do we want:
```suggestion
remove_non_numerics(pars)
```
?